src/logic/hooks/useEstimateTransactionGas.tsx

```diff
@@ -7,68 +7,52 @@ import { fromWei, toWei } from 'web3-utils'
 import { getNativeCurrency } from 'src/config'
 import {
   checkTransactionExecution,
-  estimateSafeTxGas,
-  estimateTransactionGasLimit,
+  estimateGasForTransactionExecution,
   isMaxFeeParam,
 } from 'src/logic/safe/transactions/gas'
 import { fromTokenUnit } from 'src/logic/tokens/utils/humanReadableValue'
 import { formatAmount } from 'src/logic/tokens/utils/formatAmount'
-import { calculateGasPrice, setMaxPrioFeePerGas, getFeesPerGas } from 'src/logic/wallets/ethTransactions'
+import {
+  calculateGasPrice,
+  setMaxPrioFeePerGas,
+  getFeesPerGas,
+  DEFAULT_MAX_GAS_FEE,
+  DEFAULT_MAX_PRIO_FEE,
+} from 'src/logic/wallets/ethTransactions'
 import { currentSafe } from 'src/logic/safe/store/selectors'
 import { providerSelector } from 'src/logic/wallets/store/selectors'
 import { Confirmation } from 'src/logic/safe/store/models/types/confirmation'
-import { checkIfOffChainSignatureIsPossible } from 'src/logic/safe/safeTxSigner'
 import { ZERO_ADDRESS } from 'src/logic/wallets/ethAddresses'
-import { isSpendingLimit } from 'src/routes/safe/components/Transactions/helpers/utils'
-import useCanTxExecute from './useCanTxExecute'
 
 export enum EstimationStatus {
   LOADING = 'LOADING',
   FAILURE = 'FAILURE',
   SUCCESS = 'SUCCESS',
 }
 
-export const checkIfTxIsApproveAndExecution = (
-  threshold: number,
-  txConfirmations: number,
-  txType?: string,
-  preApprovingOwner?: string,
-): boolean => {
-  if (txConfirmations === threshold) return false
-  if (!preApprovingOwner) return false
-  return txConfirmations + 1 === threshold || isSpendingLimit(txType)
-}
-
-export const checkIfTxIsCreation = (txConfirmations: number, txType?: string): boolean =>
-  txConfirmations === 0 && !isSpendingLimit(txType)
-
 type UseEstimateTransactionGasProps = {
   txData: string
   txRecipient: string
   txConfirmations?: List<Confirmation>
   txAmount?: string
-  preApprovingOwner?: string
   operation?: number
   safeTxGas?: string
-  txType?: string
   manualGasPrice?: string
   manualMaxPrioFee?: string
   manualGasLimit?: string
-  manualSafeNonce?: number // Edited nonce
```

Comment:
Can you double check if this is needed or not?

Reply:
Somehow `manualSafeNonce` was passed to the hook and used in the dependency array but nowhere else so it was essentially useless.

Reply:
You're still able to set the nonce manually though, right?

Reply:
Yes, the state for that is managed inside `EditableTxParameters` and updated in `TxModalWrapper` whenever the edit screen closes. The values inside `TxModalWrapper` are then used solely for the estimation call.